napi_js_data.md

```diff
@@ -0,0 +1,760 @@
+This API returns a N-API value corresponding to a JavaScript Array type. 
+JavaScript arrays are described in [Section 22.1](https://tc39.github.io/ecma262/#sec-array-objects) of the ECMAScript Language Specification.
+
+### *napi_create_array_with_length*
+
+#### Signature
+```
+napi_status napi_create_array(napi_env env, size_t length, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  length`: The initial length of the Array
+- `[out] result`: A `napi_value` representing a JavaScript Array
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API returns a N-API value corresponding to a JavaScript Array type. 
+The Array's length property is set to the passed in length parameter. 
+However, the underlying buffer is not guaranteed to be pre-allocated- that behavior
+is left to the underlying VM implementation. If you need the buffer to be 
+guaranteed, consider using `napi_create_external_arraybuffer`.
+
+JavaScript arrays are described in [Section 22.1](https://tc39.github.io/ecma262/#sec-array-objects) 
+of the ECMAScript Language Specification.
+
+### *napi_create_arraybuffer*
+
+#### Signature
+```
+napi_status napi_create_arraybuffer(napi_env env, size_t byte_length, void** data, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  length`: The length in bytes of the array buffer to create
+- `[out] data`: Pointer to the underlying byte buffer of the ArrayBuffer
+- `[out] result`: A `napi_value` representing a JavaScript ArrayBuffer
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API returns a N-API value corresponding to a JavaScript ArrayBuffer.
+ArrayBuffers are used to represent fixed-length binary data buffers. 
+The ArrayBuffer allocated will have an underlying byte buffer whose size is determined 
+by the `length` parameter that's passed in. 
+The underlying buffer is optionally returned back to the caller in case the caller
+wants to directly manipulate the buffer. This buffer can only be written to directly
+from Native code. To write to this buffer from JavaScript, a typed array or DataView
+object would need to be created.
+
+JavaScript ArrayBuffers are described in [Section 24.1](https://tc39.github.io/ecma262/#sec-arraybuffer-objects) 
+of the ECMAScript Language Specification.
+
+### *napi_create_buffer*
+
+#### Signature
+```
+napi_status napi_create_buffer(napi_env env, size_t size, void** data, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  size`: Size in bytes of the underlying buffer
+- `[out] data`: Raw pointer to the underlying buffer
+- `[out] result`: A `napi_value` representing a node::Buffer
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API allocates a node::Buffer object. It exists for compat with older Node versions.
```

Comment:
`node::Buffer` is still very much used (and not deprecated) in current versions of node. So I think this statement is misleading.